src/ImageSharp/Formats/Tiff/Ifd/EntryReader.cs

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Streams;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    internal class EntryReader
+    {
+        private readonly TiffStream stream;
+
+        private readonly SortedDictionary<uint, Action> extValueLoaders = new SortedDictionary<uint, Action>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EntryReader" /> class.
+        /// </summary>
+        /// <param name="stream">The stream.</param>
+        public EntryReader(TiffStream stream)
+        {
+            this.stream = stream;
+        }
+
+        public IExifValue ReadNext()
```

Comment:
I'm confused by this class. Shouldn't the existing existing `ExifReader` class be handling this?

Reply:
@JimBobSquarePants Exif is extension of Tiff and it seems that reading for Tif is a little different from reading existing `ExifReader`, I'll figure it out more and come back with an answer

Reply:
@JimBobSquarePants @brianpopow  tiff and exif have differences in tags reading, but I think it's possible to combine them (although it will probably be ugly), it will take some work, coming days I can't do, perhaps next week

Reply:
@IldarKhayrutdinov maybe you could elaborate what you want to change beforehand. If its really "ugly", we probably should not do it in the first place. Maybe we can come up with a better solution.

Reply:
@brianpopow yes, you still need to think about how to do it beautifully 🤔